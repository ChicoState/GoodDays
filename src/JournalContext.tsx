import React, { createContext, useContext, useState } from 'react';
import { JournalEntry } from './journal';

interface JournalContextType {
  journalList: JournalEntry[];
  addJournalEntry: (entry: JournalEntry) => void;
  deleteJournalEntry: (entry: JournalEntry) => void;
}

export const JournalContext = createContext<JournalContextType | undefined>(undefined);

// Dummy journal entries
const dummyEntries: JournalEntry[] = [
    {
      date: '2024-10-01',
      title: '1st Day',
      hoursActive: 4,
      hoursSleeping: 8,
      hoursFocused: 5,
      hoursOnScreen: 3,
      hoursOutside: 2,
      hoursReading: 1,
      mood: 10,
      reflection: 'Had a productive day!',
      created: new Date().toISOString(),
      updated: new Date().toISOString(),
    },
    {
      date: '2024-10-02',
      title: '2nd Day',
      hoursActive: 3,
      hoursSleeping: 7,
      hoursFocused: 4,
      hoursOnScreen: 4,
      hoursOutside: 1,
      hoursReading: 2,
      mood: 5,
      reflection: 'Just an average day.',
      created: new Date().toISOString(),
      updated: new Date().toISOString(),
    },
    {
      date: '2024-10-03',
      title: '3rd Day',
      hoursActive: 3,
      hoursSleeping: 6.5,
      hoursFocused: 2,
      hoursOnScreen: 5,
      hoursOutside: 0.5,
      hoursReading: 0,
      mood: 4,
      reflection: 'Not that good of a day.',
      created: new Date().toISOString(),
      updated: new Date().toISOString(),
    },
    {
      date: '2024-10-04',
      title: '4th Day',
      hoursActive: 1,
      hoursSleeping: 8,
      hoursFocused: 3,
      hoursOnScreen: 2,
      hoursOutside: 2,
      hoursReading: 0.5,
      mood: 9,
      reflection: 'Liesurely Day.',
      created: new Date().toISOString(),
      updated: new Date().toISOString(),
    },
    {
      date: '2024-10-05',
      title: '5th Day',
      hoursActive: 0.5,
      hoursSleeping: 8,
      hoursFocused: 7,
      hoursOnScreen: 3,
      hoursOutside: 0.5,
      hoursReading: 1,
      mood: 7,
      reflection: 'Focused Day.',
      created: new Date().toISOString(),
      updated: new Date().toISOString(),
    },
  ];

export const useJournal = () => {
  const context = useContext(JournalContext);
  if (!context) {
    throw new Error('useJournal must be used within a JournalProvider');
  }
  return context;
};

export const JournalProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  const [journalEntries, setJournalEntries] = useState<JournalEntry[]>(dummyEntries);

  const addJournalEntry = (entry: JournalEntry) => {

    setJournalEntries(prevEntries => {
      const existingEntryIndex = prevEntries.findIndex(e => e.date === entry.date);

      if (existingEntryIndex !== -1) {
        // Replace the existing entry with the new one
        const updatedEntries = [...prevEntries];
        updatedEntries[existingEntryIndex] = entry;
        return updatedEntries;
      } else {
        // Add the new entry to the end
        return [...prevEntries, entry];
      }
    });
  };

  const deleteJournalEntry = (entry: JournalEntry) => {

    setJournalEntries( entries => {
      // find entry with same name created on the same day
      const index = entries.findIndex(event => event.date === entry.date && event.title === entry.title);
      
      // remove match if found
      if (index !== -1) {
          return entries.filter((_, i) => i !== index);
      } else {
          console.log("Event not found");
      }

    })
  }

  return (
    <JournalContext.Provider value={{ journalList: journalEntries, addJournalEntry, deleteJournalEntry }}>
      {children}
    </JournalContext.Provider>
  );
};
