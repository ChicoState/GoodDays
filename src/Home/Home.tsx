import { JournalEntry } from "../journal";
import { useJournal } from "../JournalContext"
import { Link } from "react-router-dom";
import { useState } from "react";
import { useContext } from "react";

// ceverett, Sprint in Progress #2: (calendar functionality)
import Calendar from 'react-calendar'; // "npm install,list react-calendar"
import 'react-calendar/dist/Calendar.css'; // (optional) default styles

type SortDir = -1 | 1;

/*
 * creates a sorting function for sorting journal entryes by date in either
 * ascending or descending order
 */
function get_sort(dir: SortDir) {
    return (a: JournalEntry, b: JournalEntry) => {
        if (a.date === b.date) {
            return 0;
        } else if (a.date < b.date) {
            return -1 * dir;
        } else {
            return 1 * dir;
        }
    };
}

const sort_desc = get_sort(-1);

interface EntriesViewProps {
    entries: JournalEntry[]
}

const EntriesView = () => {
    let tr_rows = [];

    const { journalList, deleteJournalEntry } = useJournal();

    const onDeleteEntry = (entry:JournalEntry) => {
        deleteJournalEntry(entry);
    }

    // this is probably not optimal
    for (let entry of journalList.sort(sort_desc)) {
        tr_rows.push(
            <tr key={entry.date}>
                <td><Link to="/Create" state={{entry}}>{entry.date}</Link></td>
                <td>{entry.title}</td>
                <td>{entry.hoursActive}</td>
                <td>{entry.hoursSleeping}</td>
                <td>{entry.hoursFocused}</td>
                <td>{entry.hoursOnScreen}</td>
                <td>{entry.hoursOutside}</td>
                <td>{entry.hoursReading}</td>
                <td>{entry.mood}</td>
                <td>{entry.reflection}</td>
                <td>{entry.created}</td>
                <td>{entry.updated}</td>
                <td> <button onClick={() => onDeleteEntry(entry)}>Delete &#x274C;</button></td>
            </tr>
        );
    }

    if (tr_rows.length === 0) {
        tr_rows.push(<tr key={"empty"}>
            <td colSpan={12}>No Entries</td>
        </tr>);
    }

    return (
        <div>
            <table>
                <thead>
                    <tr>
                        <th>Date</th>
                        <th>Title</th>
                        <th>Hours Active</th>
                        <th>Hours Sleeping</th>
                        <th>Hours Focused</th>
                        <th>Hours on Screen</th>
                        <th>Hours Outside</th>
                        <th>Hours Reading</th>
                        <th>Mood</th>
                        <th>Contents</th>
                        <th>Created</th>
                        <th>Updated</th>
                    </tr>
                </thead>
                <tbody>{tr_rows}</tbody>
            </table>
        </div>
    );
};

type HomeProps = {};

const Home = ({}: HomeProps) => {


    return (
    
    <><div>
        
        <h1>Moods Calendar for My GoodDays</h1>
        <Calendar />
        
        </div><br /><div>
            
            <EntriesView/>
            
            </div></>
    );
}

export default Home;