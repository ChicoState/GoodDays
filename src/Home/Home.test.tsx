import * as React from "react";
import{render, screen} from "@testing-library/react";
import Home from "./Home";
import { JournalProvider } from "../JournalContext";
import { HashRouter } from "react-router-dom";
import "@testing-library/jest-dom";

describe("Home renders correctly", () => {

    test("displays calendar and headers", () => {
        render(<JournalProvider><HashRouter><Home /></HashRouter></JournalProvider>)

    expect(screen.getByText(/Moods Calendar for My GoodDays/i)).toBeInTheDocument();

    //get cal element:
    //const calElement = document.querySelector('.react-calendar'); //queryselecor looks for specific element in document
    const testCalendar = document.querySelector('.calendar'); //queryselecor looks for specific element in document
    //const testContainer = document.querySelector('.container');
    //const calElement = document.querySelector('.container');
    
    expect(testCalendar).toBeInTheDocument();
    });





    // test("calendar container is found", () => {

    //     const { container } = render(<JournalProvider><HashRouter><Home /></HashRouter></JournalProvider>)

    //     const testContainer = container.querySelector('.container');

    //     expect(testContainer).toBeInTheDocument();
    // });

});

/*
describe("Home", () => {
    test("matches snapshot", () => { // need to change this because data changes, so snapshot probably wont work
       const{asFragment} = render(<JournalProvider><MemoryRouter><Home></Home></MemoryRouter></JournalProvider>);
       expect(asFragment()).toMatchSnapshot();
    });
});
*/
