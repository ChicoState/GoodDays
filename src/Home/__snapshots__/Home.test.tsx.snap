// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Moods Calendar for My GoodDays
    </h1>
    <div
      class="react-calendar"
    >
      <div
        class="react-calendar__navigation"
      >
        <button
          aria-label=""
          class="react-calendar__navigation__arrow react-calendar__navigation__prev2-button"
          type="button"
        >
          «
        </button>
        <button
          aria-label=""
          class="react-calendar__navigation__arrow react-calendar__navigation__prev-button"
          type="button"
        >
          ‹
        </button>
        <button
          aria-label=""
          class="react-calendar__navigation__label"
          style="flex-grow: 1;"
          type="button"
        >
          <span
            class="react-calendar__navigation__label__labelText react-calendar__navigation__label__labelText--from"
          >
            December 2024
          </span>
        </button>
        <button
          aria-label=""
          class="react-calendar__navigation__arrow react-calendar__navigation__next-button"
          type="button"
        >
          ›
        </button>
        <button
          aria-label=""
          class="react-calendar__navigation__arrow react-calendar__navigation__next2-button"
          type="button"
        >
          »
        </button>
      </div>
      <div
        class="react-calendar__viewContainer"
      >
        <div
          class="react-calendar__month-view"
        >
          <div
            style="display: flex; align-items: flex-end;"
          >
            <div
              style="flex-grow: 1; width: 100%;"
            >
              <div
                class="react-calendar__month-view__weekdays"
                style="display: flex; flex-wrap: nowrap;"
              >
                <div
                  class="react-calendar__month-view__weekdays__weekday"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                >
                  <abbr
                    aria-label="Monday"
                    title="Monday"
                  >
                    Mon
                  </abbr>
                </div>
                <div
                  class="react-calendar__month-view__weekdays__weekday"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                >
                  <abbr
                    aria-label="Tuesday"
                    title="Tuesday"
                  >
                    Tue
                  </abbr>
                </div>
                <div
                  class="react-calendar__month-view__weekdays__weekday"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                >
                  <abbr
                    aria-label="Wednesday"
                    title="Wednesday"
                  >
                    Wed
                  </abbr>
                </div>
                <div
                  class="react-calendar__month-view__weekdays__weekday"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                >
                  <abbr
                    aria-label="Thursday"
                    title="Thursday"
                  >
                    Thu
                  </abbr>
                </div>
                <div
                  class="react-calendar__month-view__weekdays__weekday"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                >
                  <abbr
                    aria-label="Friday"
                    title="Friday"
                  >
                    Fri
                  </abbr>
                </div>
                <div
                  class="react-calendar__month-view__weekdays__weekday react-calendar__month-view__weekdays__weekday--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                >
                  <abbr
                    aria-label="Saturday"
                    title="Saturday"
                  >
                    Sat
                  </abbr>
                </div>
                <div
                  class="react-calendar__month-view__weekdays__weekday react-calendar__month-view__weekdays__weekday--current react-calendar__month-view__weekdays__weekday--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                >
                  <abbr
                    aria-label="Sunday"
                    title="Sunday"
                  >
                    Sun
                  </abbr>
                </div>
              </div>
              <div
                class="react-calendar__month-view__days"
                style="display: flex; flex-wrap: wrap;"
              >
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 25, 2024"
                  >
                    25
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 26, 2024"
                  >
                    26
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 27, 2024"
                  >
                    27
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 28, 2024"
                  >
                    28
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 29, 2024"
                  >
                    29
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="November 30, 2024"
                  >
                    30
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 1, 2024"
                  >
                    1
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 2, 2024"
                  >
                    2
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 3, 2024"
                  >
                    3
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 4, 2024"
                  >
                    4
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 5, 2024"
                  >
                    5
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 6, 2024"
                  >
                    6
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 7, 2024"
                  >
                    7
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__tile--now react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 8, 2024"
                  >
                    8
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 9, 2024"
                  >
                    9
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 10, 2024"
                  >
                    10
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 11, 2024"
                  >
                    11
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 12, 2024"
                  >
                    12
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 13, 2024"
                  >
                    13
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 14, 2024"
                  >
                    14
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 15, 2024"
                  >
                    15
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 16, 2024"
                  >
                    16
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 17, 2024"
                  >
                    17
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 18, 2024"
                  >
                    18
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 19, 2024"
                  >
                    19
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 20, 2024"
                  >
                    20
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 21, 2024"
                  >
                    21
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 22, 2024"
                  >
                    22
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 23, 2024"
                  >
                    23
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 24, 2024"
                  >
                    24
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 25, 2024"
                  >
                    25
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 26, 2024"
                  >
                    26
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 27, 2024"
                  >
                    27
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 28, 2024"
                  >
                    28
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 29, 2024"
                  >
                    29
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 30, 2024"
                  >
                    30
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="December 31, 2024"
                  >
                    31
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="January 1, 2025"
                  >
                    1
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="January 2, 2025"
                  >
                    2
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="January 3, 2025"
                  >
                    3
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="January 4, 2025"
                  >
                    4
                  </abbr>
                </button>
                <button
                  class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                  style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                  type="button"
                >
                  <abbr
                    aria-label="January 5, 2025"
                  >
                    5
                  </abbr>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <br />
  <div>
    <table>
      <thead>
        <tr>
          <th>
            Date
          </th>
          <th>
            Title
          </th>
          <th>
            Hours Active
          </th>
          <th>
            Hours Sleeping
          </th>
          <th>
            Hours Focused
          </th>
          <th>
            Hours on Screen
          </th>
          <th>
            Hours Outside
          </th>
          <th>
            Hours Reading
          </th>
          <th>
            Mood
          </th>
          <th>
            Contents
          </th>
          <th>
            Created
          </th>
          <th>
            Updated
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <a
              href="/Create"
            >
              2024-10-02
            </a>
          </td>
          <td>
            Second Day
          </td>
          <td>
            3
          </td>
          <td>
            7
          </td>
          <td>
            4
          </td>
          <td>
            4
          </td>
          <td>
            1
          </td>
          <td>
            2
          </td>
          <td>
            1
          </td>
          <td>
            Just an average day.
          </td>
          <td>
            2024-12-09T02:22:46.935Z
          </td>
          <td>
            2024-12-09T02:22:46.935Z
          </td>
          <td>
             
            <button>
              Delete ❌
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/Create"
            >
              2024-10-01
            </a>
          </td>
          <td>
            First Day
          </td>
          <td>
            4
          </td>
          <td>
            8
          </td>
          <td>
            5
          </td>
          <td>
            3
          </td>
          <td>
            2
          </td>
          <td>
            1
          </td>
          <td>
            10
          </td>
          <td>
            Had a productive day!
          </td>
          <td>
            2024-12-09T02:22:46.934Z
          </td>
          <td>
            2024-12-09T02:22:46.934Z
          </td>
          <td>
             
            <button>
              Delete ❌
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
